### PR TITLE
Datum transform UX improvements

### PR DESCRIPTION
**Pull request by 3nids**
* use table widget instead of tree widget in the datum transform dialog for better column display
* more room for the proj selection widget
* fix dialog shown when option is deactivated

### CODE REVIEW

File: src/gui/qgsdatumtransformdialog.cpp [hunk 1]
```diff
@@ -59,32 +64,30 @@ QgsDatumTransformDialog::QgsDatumTransformDialog( const QgsCoordinateReferenceSy
...
+
+  int row = 0;
 
   for ( const QgsCoordinateTransform::TransformPair &transform : qgis::as_const( mDatumTransforms ) )
   {
-    QTreeWidgetItem *item = new QTreeWidgetItem();
     bool itemDisabled = false;
     bool itemHidden = false;
 
+    if ( transform.sourceTransformId == -1 || transform.destinationTransformId == -1 )
```
Comment (3nids) [inline src/gui/qgsdatumtransformdialog.cpp]: @nyalldawson the code was like that: do not show the line if one of the transform is invalid. Does it make sense? Isn't there case where there would be only one transform? For instance if the dest or source is already on WGS84???

Comment (nyalldawson) [reply src/gui/qgsdatumtransformdialog.cpp]: Yes I think that condition is wrong

Comment (3nids) [reply src/gui/qgsdatumtransformdialog.cpp]: ok fixed

**Comment by 3nids** [comment]
also wondering why there is twice the CHENYX06a.gsb (once as source, once as dest).
Is `QgsCoordinateTransform::datumTransformations` behaving as expected?
I think we should have it only once.

screen shot 2017-12-18 at 15 31 36

**Comment by nyalldawson** [comment]
Can you rebase on master and let me know if you still see that? I've pushed a bunch of fixes

**Comment by nyalldawson** [comment]
Actually I think it's a typo. Can you change QgsCoordinateTransform.cpp:753 to reverseDirectTransforms instead and see if it works?

**Comment by 3nids** [comment]
building....

**Comment by 3nids** [comment]
@nyalldawson indeed the typo fixed it

**Comment by nyalldawson** [comment]
Thanks! I'll see about adding some unit tests for these methods. Just gotta work out some transforms which should be always available to test with...

**Comment by 3nids** [comment]
@nyalldawson it works but made the TestQgsCoordinateTransform failed. Any idea?

**Comment by nyalldawson** [comment]
@3nids https://github.com/qgis/QGIS/pull/5906 should fix it

**Comment by 3nids** [comment]
thanks, waiting for your merge then!

**Comment by nirvn** [comment]
@nyalldawson , @3nids , great job with those follow up tweaks n fixes.

**Pull request merged**